File: src/java/ai/nektron/grownet/metrics/RegionMetrics.java
package ai.nektron.grownet.metrics;

public class RegionMetrics {
    private long deliveredEvents;
    private long totalSlots;
    private long totalSynapses;

    private long   activePixels;
    private double centroidRow;
    private double centroidCol;
    private int    bboxRowMin;
    private int    bboxRowMax;
    private int    bboxColMin;
    private int    bboxColMax;

    public long getDeliveredEvents() { return deliveredEvents; }
    public long getTotalSlots()      { return totalSlots; }
    public long getTotalSynapses()   { return totalSynapses; }
    public long getActivePixels()    { return activePixels; }
    public double getCentroidRow()   { return centroidRow; }
    public double getCentroidCol()   { return centroidCol; }
    public int getBboxRowMin()       { return bboxRowMin; }
    public int getBboxRowMax()       { return bboxRowMax; }
    public int getBboxColMin()       { return bboxColMin; }
    public int getBboxColMax()       { return bboxColMax; }

    public RegionMetrics incDeliveredEvents()         { deliveredEvents++; return this; }
    public RegionMetrics addSlots(long n)             { totalSlots     += n; return this; }
    public RegionMetrics addSynapses(long n)          { totalSynapses  += n; return this; }

    public RegionMetrics setActivePixels(long n) { this.activePixels = n; return this; }
    public RegionMetrics setCentroid(double row, double col) { this.centroidRow = row; this.centroidCol = col; return this; }
    public RegionMetrics setBBox(int rowMin, int rowMax, int colMin, int colMax) {
        this.bboxRowMin = rowMin; this.bboxRowMax = rowMax; this.bboxColMin = colMin; this.bboxColMax = colMax; return this;
    }
}

File: src/typescript/grownet-ts/src/metrics/RegionMetrics.ts
export class RegionMetrics {
  private deliveredEvents: number;
  private totalSlots: number;
  private totalSynapses: number;
  private activePixels: number;
  private centroidRow: number;
  private centroidCol: number;
  private bboxRowMin: number;
  private bboxRowMax: number;
  private bboxColMin: number;
  private bboxColMax: number;

  constructor(
    deliveredEvents: number,
    totalSlots: number,
    totalSynapses: number,
    activePixels: number,
    centroidRow: number,
    centroidCol: number,
    bboxRowMin: number,
    bboxRowMax: number,
    bboxColMin: number,
    bboxColMax: number,
  ) {
    this.deliveredEvents = deliveredEvents;
    this.totalSlots = totalSlots;
    this.totalSynapses = totalSynapses;
    this.activePixels = activePixels;
    this.centroidRow = centroidRow;
    this.centroidCol = centroidCol;
    this.bboxRowMin = bboxRowMin;
    this.bboxRowMax = bboxRowMax;
    this.bboxColMin = bboxColMin;
    this.bboxColMax = bboxColMax;
  }

  getDeliveredEvents(): number { return this.deliveredEvents; }
  getTotalSlots(): number { return this.totalSlots; }
  getTotalSynapses(): number { return this.totalSynapses; }
  getActivePixels(): number { return this.activePixels; }
  getCentroidRow(): number { return this.centroidRow; }
  getCentroidCol(): number { return this.centroidCol; }
  getBboxRowMin(): number { return this.bboxRowMin; }
  getBboxRowMax(): number { return this.bboxRowMax; }
  getBboxColMin(): number { return this.bboxColMin; }
  getBboxColMax(): number { return this.bboxColMax; }
}
export class RegionMetrics {
